documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -1024,4 +1029,180 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
+
+**Description**: Each OpenAPI json document must contain a security definitions section and the section must adhere to a certain format.
+
+**Why the rule is important**: Missing security definitions section does not describe the Azure services security model accurately.
+
+**How to fix the violation**: Ensure that the document has a security definition section as described in [https://github.com/Azure/autorest/tree/master/docs/developer/validation-rules/security-definitions-structure-validation.md](https://github.com/Azure/autorest/tree/master/docs/developer/validation-rules/security-definitions-structure-validation.md)
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2023" />R2023 SummaryAndDescriptionMustNotBeSame
+**Output Message**: The summary and description values should not be same.
+
+**Description**: Each operation has a summary and description values. They must not be same.
+
+**Why the rule is important**: The summary must provide a short summary of the operation. The description must provide a detailed description of the operation. This will ensure that all the operations are well documented. 
+
+**How to fix the violation**: Provide a short summary for the summary section and a detailed description for the description section.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2044" />R2044 InvalidVerbUsed
+**Output Message**: Permissible values for HTTP Verb are delete,get,put,patch,head,options,post.
+
+**Description**: Each operation definition must have a HTTP verb and it must be delete/get/put/patch/head/options/post.
+
+**Why the rule is important**: Delete/get/put/patch/head/options/post are the only valid HTTP operations.
+
+**How to fix the violation**: Provide a correct HTTP verb.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R4000-2" />R4000 ParameterDescriptionRequired
+**Output Message**: {0} lacks 'description' property. Consider adding a 'description' element. Accurate description is essential for maintaining reference documentation.
+
+**Description**: A parameter must have 'description' property.
+
+**Why the rule is important**: This will ensure that the parameter is well documented.
+
+**How to fix the violation**: For each parameter, provide a 'description' property.
```

Comment:
How can we provide a generic description so the customer can copy/paste. The idea was that the descrption should be detailed for the sepcific scenario